draft-ietf-ppm-dap.md

```diff
@@ -639,9 +639,12 @@ struct {
 struct {} Empty;
 ~~~
 
-DAP uses the 16-byte `ReportID` as the nonce parameter for the VDAF
-`shard` and `prep_init` methods (see {{!VDAF, Section 5}}). Thus for a VDAF to
-be compatible with DAP, it MUST specify a `NONCE_SIZE` of 16 bytes.
+DAP uses the 16-byte `ReportID` as the nonce parameter for the VDAF `shard` and
+`prep_init` methods (see {{!VDAF, Section 5}}). Additionally, DAP includes
+messages defined in the VDAF specification encoded as opaque byte strings within
+various DAP messages. Thus, for a VDAF to be compatible with DAP, it MUST
+specify a `NONCE_SIZE` of 16 bytes, and must specify encodings for all relevant
+messages into byte strings.
```

Comment:
Can we enumerate what these messages are? It's input shares, prepare shares and aggregate shares, right?

Reply:
We can, with the downside that we're now on the hook to keep this list up-to-date. IMO, we might be better served by leaving the list ambiguous.

Note that the last two types (`PrepShare` and `PrepMessage`) require an encoding because of their inclusion in ping-pong messages, the serialization of which is specified in VDAF (which itself assumes that these types have a defined encoding). I think it's less confusing to list the specific VDAF types, since that is what someone trying to implement DAP would be able to reference when looking at any given VDAF.

Reply:
Any thoughts on mentioning encodability of `PrepState`, either in this list or separately, given the considerations I mention above?

Reply:
> I think it's less confusing to list the specific VDAF types, since that is what someone trying to implement DAP would be able to reference when looking at any given VDAF.

Yep, I think this is the right trade-off.

> Any thoughts on mentioning encodability of `PrepState`, either in this list or separately, given the considerations I mention above?

I don't think we need to discuss it in this protocol. I agree that any practical DAP implementation will need to be able to somehow encode and store `PrepState`, but since that representation doesn't have to be agreed upon by more than one implementation, this document doesn't need to be prescriptive about it.

Reply:
Note that we need an encoding for the aggregation parameter as well, in collection requests and responses.

Reply:
Gah, thanks, I mistakenly conflated AggParam with AggShare when I was generating the list.